nep-11.mediawiki

```diff
@@ -0,0 +1,418 @@
+===Optional methods===
+
+====tokens====
+
+<pre>
+{
+  "name": "tokens",
+  "parameters": [],
+  "returntype": "InteropInterface<iterator>"
+}
+</pre>
+
+Returns an <code>iterator</code> that contains all of the tokens minted by the contract.
+
+====properties====
+
+<pre>
+{
+  "name": "properties",
+  "parameters": [
+    {
+      "name": "tokenid",
+      "type": "ByteString"
+    }
+  ],
+  "returntype": "Map"
+}
+</pre>
+
+Returns a serialized NVM object containing the properties for the given NFT.
+The NVM object must conform to the "NEO NFT Metadata JSON Schema".
+
+The parameter <code>tokenid</code> SHOULD be a valid NFT.
+If not, this method SHOULD <code>throw</code> an exception.
+
+Following is a description of the JSON Schema:
+<pre>
+{
+  "title": "Asset Metadata",
+  "type": "object",
+  "required": ["name"],
+  "properties": {
+    "name": {
+      "type": "string",
+      "description": "Identifies the asset to which this NFT represents."
+    },
+    "description": {
+      "type": "string",
+      "description": "Optional. Describes the asset to which this NFT represents."
+    },
+    "image": {
+      "type": "string",
+      "format": "uri",
+      "description": "Optional. A URI pointing to a resource with mime type image/* representing the asset to which this NFT represents. Consider making any images at a width between 320 and 1080 pixels and aspect ratio between 1.91:1 and 4:5 inclusive."
+    },
+    "tokenURI": {
+      "type": "string",
+      "format": "uri",
+      "description": "Optional. A distinct URI for a given asset that adheres to RFC 3986"
+    }
+  }
+}
+</pre>
+
+Given this schema, the simplest json object that can be constructed is the following:
+<pre>
+{
+   "name": "Slime 1"
+}
+</pre>
+
+And if all the properties are included:
+<pre>
+{
+   "name": "Slime 2",
+   "description": "A slime",
+   "image": "{some image URI}",
+   "tokenURI": "{some URI}"
+}
+</pre>
+
+===Receiver methods===
+
+====onNEP11Payment====
+
+<pre>
+{
+  "name": "onNEP11Payment",
+  "parameters": [
+    {
+      "name": "from",
+      "type": "Hash160"
+    },
+    {
+      "name": "amount",
+      "type": "Integer"
+    },
+    {
+      "name": "tokenid",
+      "type": "ByteString"
+    }
+  ],
+  "returntype": "Void"
+}
+</pre>
+
+If a contract is to receive NFTs, it must implement the receiver method. Transferring NFTs to any contract that does not implement the receiver method will cause the transaction to fail. If the receiver doesn't want to receive this transfer it MUST abort the execution.
+
+The meaning of the parameters of the <code>onNEP11Payment</code> method is consistent with that of <code>transfer</code>. 
+
+===Events===
+
+====Transfer====
+
+<pre>
+{
+  "name": "Transfer",
+  "parameters": [
+    {
+      "name": "from",
+      "type": "Hash160"
+    },
+    {
+      "name": "to",
+      "type": "Hash160"
+    },
+    {
+      "name": "amount",
+      "type": "Integer"
+    },
+    {
+      "name": "tokenid",
```

Comment:
We implement some checks in `neo-go` and currently actual parameter name in code (`tokenId` doesn't match `tokenid` here). Can we unify both?

Reply:
We can unify them in native contracts. But we have no way to guarantee that other contracts use the same parameter names. And the parameter name is not important, it does not affect anything.

Reply:
Updated.